dowhy/gcm/distribution_change.py

```diff
@@ -88,11 +88,34 @@ def mechanism_change_test(
         return conditional_independence_test(joint_target_samples, data_set_indices, joint_parent_data)
 
 
+def remove_invariant_nodes(
+    invariant_nodes: list,
+    causal_model: ProbabilisticCausalModel,
+    old_data: pd.DataFrame,
+    auto_assignment_quality: Optional[AssignmentQuality],
+):
+    for invar_node in invariant_nodes:
+        if invar_node in list(causal_model.graph.nodes):
+            # Get parent and child nodes
+            parents = get_ordered_predecessors(causal_model.graph, invar_node)
+            children = list(causal_model.graph.successors(invar_node))
+            # Remove the middle node
+            causal_model.graph.remove_node(invar_node)
+            # Connect parent and child nodes
+            for parent in parents:
+                for child in children:
+                    causal_model.graph.add_edge(parent, child)
+            # Update the causal mechanism for the child nodes
+            for child in children:
+                set_causal_mechanism_for_node(causal_model, child, old_data, quality=auto_assignment_quality)
+
+
 def distribution_change(
     causal_model: ProbabilisticCausalModel,
     old_data: pd.DataFrame,
     new_data: pd.DataFrame,
     target_node: Any,
+    invariant_nodes: list = None,
```

Comment:
Similar as above: The type for `invariant_nodes` should be `List[Any]` for consistency.